### ISSUE TITLE
cloud_firestore crashes on reopening Flutter app 

### ISSUE THREAD

**Issue by adifyr** [FirebaseExtended/flutterfire#2625]
**Describe the bug**
My flutter app crashes with an internal `NullPointerException` whenever I leave the app using the Android System Back Button, re-enter and try to do a `FieldValue.arrayUnion()` or `FieldValue.arrayRemove()`. Firestore reads seem to be working fine.

**To Reproduce**
1. Open the app.
2. Close it using the device's system back button.
4. On Re-entry attempt to do an `docRef.updateData({'key': FieldValue.arrayRemove([value])})`.

**Expected behavior**
The app isn't supposed to crash and do the document update as expected.

**Additional context**
cloud_firestore version: ^0.13.5

Here are the logs:
```
E/AndroidRuntime(11095): java.lang.NullPointerException: Attempt to invoke virtual method 'void io.flutter.plugin.common.MethodChannel.invokeMethod(java.lang.String, java.lang.Object)' on a null object reference
E/AndroidRuntime(11095): 	at io.flutter.plugins.firebase.cloudfirestore.CloudFirestorePlugin$EventObserver.onEvent(CloudFirestorePlugin.java:451)
E/AndroidRuntime(11095): 	at io.flutter.plugins.firebase.cloudfirestore.CloudFirestorePlugin$EventObserver.onEvent(CloudFirestorePlugin.java:433)
E/AndroidRuntime(11095): 	at com.google.firebase.firestore.Query.lambda$addSnapshotListenerInternal$2(com.google.firebase:firebase-firestore@@21.3.0:1025)
E/AndroidRuntime(11095): 	at com.google.firebase.firestore.Query$$Lambda$3.onEvent(Unknown Source:6)
E/AndroidRuntime(11095): 	at com.google.firebase.firestore.core.AsyncEventListener.lambda$onEvent$0(com.google.firebase:firebase-firestore@@21.3.0:42)
E/AndroidRuntime(11095): 	at com.google.firebase.firestore.core.AsyncEventListener$$Lambda$1.run(Unknown Source:6)
E/AndroidRuntime(11095): 	at android.os.Handler.handleCallback(Handler.java:790)
E/AndroidRuntime(11095): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime(11095): 	at android.os.Looper.loop(Looper.java:164)
E/AndroidRuntime(11095): 	at android.app.ActivityThread.main(ActivityThread.java:6626)
E/AndroidRuntime(11095): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(11095): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
E/AndroidRuntime(11095): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:811)
```

**Flutter doctor**
```
[√] Flutter (Channel beta, 1.18.0-11.1.pre, on Microsoft Windows [Version 10.0.18363.836], locale en-US)
    • Flutter version 1.18.0-11.1.pre at C:\src\flutter
    • Framework revision 2738a1148b (11 days ago), 2020-05-13 15:24:36 -0700
    • Engine revision ef9215ceb2
    • Dart version 2.9.0 (build 2.9.0-8.2.beta)


[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\adity\AppData\Local\Android\sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 3.6)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.8052
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] VS Code (version 1.45.1)
    • VS Code at C:\Users\adity\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.10.2

[√] Connected device (3 available)
    • Moto G 5 Plus • ZY2242FXF8 • android-arm    • Android 8.1.0 (API 27)
    • Web Server    • web-server • web-javascript • Flutter Tools
    • Chrome        • chrome     • web-javascript • Google Chrome 81.0.4044.138

• No issues found!

```

**Comment by iapicca** [comment]
Hi @adifyr 
can you please provide 
your `flutter run --verbose`,
and a reproducible minimal code sample.
Thank you

**Comment by Salakar** [comment]
Hey all, as part of our on-going work for #2582, this has been resolved in our Firebase Firestore rework (#2913) - which has now been merged into master. We'll look at publishing some prereleases in the next few days. Thank you
